notify/opsgenie/opsgenie.go

```diff
@@ -297,3 +272,50 @@ func (n *Notifier) createRequests(ctx context.Context, as ...*types.Alert) ([]*h
 
 	return requests, true, nil
 }
+
+func (n *Notifier) updateAlert(ctx context.Context, tmpl func(string) string, key notify.Key, alias string, requests *[]*http.Request) (bool, error) {
+	message, truncated := notify.TruncateInRunes(tmpl(n.conf.Message), maxMessageLenRunes)
+	if truncated {
+		level.Warn(n.logger).Log("msg", "Truncated message", "alert", key, "max_runes", maxMessageLenRunes)
+	}
+
+	updateMessageEndpointURL := n.conf.APIURL.Copy()
+	updateMessageEndpointURL.Path += fmt.Sprintf("v2/alerts/%s/message", alias)
+	q := updateMessageEndpointURL.Query()
+	q.Set("identifierType", "alias")
+	updateMessageEndpointURL.RawQuery = q.Encode()
+	updateMsgMsg := &opsGenieUpdateMessageMessage{
+		Message: message,
+	}
+	var updateMessageBuf bytes.Buffer
+	if err := json.NewEncoder(&updateMessageBuf).Encode(updateMsgMsg); err != nil {
+		return false, err
+	}
+	req, err := http.NewRequest("PUT", updateMessageEndpointURL.String(), &updateMessageBuf)
+	if err != nil {
+		return true, err
```

Comment:
I'm not quite sure why an error from NewRequest() returned should be retried. Looking at the [source code](https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/net/http/request.go;l=838-854), we should probably not retry.

Reply:
I did not change the existing logic here. I will change it